src/main/webapp/blur.js

```diff
@@ -115,3 +117,235 @@ function getImageWithBlurredAreas(image, blurRadius) {
+  const blurredCtx = blurredCanvas.getContext('2d');
+  const blurredData = blurredCtx.createImageData(originalData);
+
+  for (let y = 0; y < originalData.height; ++y) {
+    for (let x = 0; x < originalData.width; ++x) {
+      // the blurred color of the pixel we are looking at.
+      const newColor = new Color(0, 0, 0, 0);
+
+      // should we blur or smooth current pixel.
+      let curKernel = kernel;
+      if (y < smoothSizes.top ||
+          originalData.height - y < smoothSizes.bottom ||
+          x < smoothSizes.left ||
+          originalData.width - x < smoothSizes.right) {
+        curKernel = smoothKernel;
+      }
+
+      // if we have to look at the pixel which is out of borders of
+      // data provided to us - use original pixel color.
+      const defaultColor = getColorForCoord(x, y, originalData);
+
+      for (let i = 0; i < curKernel.length; ++i) {
+        for (let j = 0; j < curKernel.length; ++j) {
+          const halfKernelLen = Math.floor(curKernel.length / 2);
+
+          const curColor = getColorForCoord(x + j - halfKernelLen,
+              y + i - halfKernelLen, originalData, defaultColor);
+
+          curColor.multiple(curKernel[i][j]);
+          newColor.add(curColor);
+        }
+      }
+
+      // colors are stored in ImageData in RGBA format in
+      // one-dimentional array. Get the index of red color
+      // in that array. Then green color will have red + 1 index,
+      // etc.
+      const redIndex = y * (blurredData.width * 4) + x * 4;
+      blurredData.data[redIndex] = newColor.red;
+      blurredData.data[redIndex + 1] = newColor.green;
+      blurredData.data[redIndex + 2] = newColor.blue;
+      blurredData.data[redIndex + 3] = newColor.alpha;
+    }
+  }
+
+  return blurredData;
+}
+
+/**
+ * Function to get image with some areas blurred.
+ * We use our own implementation of blurring
+ * algorithm here. The idea is that we construct some
+ * squared matrix called kernel, then we move that
+ * kernel through our image pixels and recalculate
+ * each pixel as kernel_matrix * matrix_of_pixels_around_ours.
+ * We also apply smooth edges by blurring some place around
+ * rectangles with less blurRadius.
+ * @param {ImageObject} image
+ * @param {Number} blurRadius
+ * @return {ImageObject}
+ */
+function getImageWithBlurredByUsAreas(image, blurRadius) {
+  const kernelSize = blurRadius;
+  const kernel = getKernel(kernelSize);
+
+  // the less kernelSize - the less blur will be applied.
+  const smoothKernel = getKernel(kernelSize / 2);
+
+  const originalCanvas = createCanvasForImage(image.object);
+  const originalCtx = originalCanvas.getContext('2d');
+  originalCtx.drawImage(image.object, 0, 0);
+
+  const blurredCanvas = createCanvasForImage(image.object);
+  const blurredCtx = blurredCanvas.getContext('2d');
+  blurredCtx.drawImage(image.object, 0, 0);
+
+  for (const rect of image.blurAreas) {
+    if (!rect.toBeBlurred) {
+      continue;
+    }
+
+    const smoothEdgesSizes = {
```

Comment:
Add a comment here to explain the number 7. I guess it is some empirical value you found, so you can say so in a comment.